### DISCUSSION TITLE
Plugin and menu issues

### DISCUSSION THREAD

**Discussion by goclimbing** [bootscore/bootscore#896] [Q&A]
A search didn't result in anything but there are no links in the menu on mobile.  Did I miss a setup step?

Plugins:
SmartSlider 3
something up with the z-index maybe?  
![Screenshot 2024-12-13 at 4 25 15 PM](https://github.com/user-attachments/assets/276bc611-fc62-4196-a755-382f1a6d85a5)

All search bars that drop down end up behind the hero slider on my home page.

WBW Product Filter PRO - same problem with the z-indexing.  

Also, instead of updating the page with the filtered products it reloads the page and leaves it blank with no filter or products (though it does show total search results).

![Screenshot 2024-12-13 at 4 33 02 PM](https://github.com/user-attachments/assets/434d31b7-3c33-46bf-b601-6ded955d852c)

![Screenshot 2024-12-13 at 2 58 51 PM](https://github.com/user-attachments/assets/879f03ef-0db1-4e0a-a5d8-8e578ed8e067)

**Comment by goclimbing** [comment]
Is there a way to use a different logo for mobile built into this theme?  menu, cart, user account and search button are pushed to second line.

**Reply by goclimbing** [reply]
https://www.tradingview.com/ iframe widgets don't update like they are suppose to.

Giving this theme a real test drive ;) 

Here's the URL: 
https://psbullion.kinsta.cloud/shop/

it's suppose to be a secret for a few more weeks ;)

**Reply by crftwrk** [reply]
> Is there a way to use a different logo for mobile built into this theme? menu, cart, user account and search button are pushed to second line.

For now, only way is to copy `header.php` to your child and code it there. But we're working on an update that adds dozens of actions and a lot more filters, see https://github.com/bootscore/bootscore/pull/841. When this is ready, you can add a mobile logo without overriding `header.php` in child.

**Comment by crftwrk** [comment]
> A search didn't result in anything but there are no links in the menu on mobile. Did I miss a setup step?

You are using white nav-links on white offcanvas bg. Change offcanvas bg or nav-link colors in mobile view.

<img width="461" alt="Bildschirmfoto 2024-12-14 um 12 36 47" src="https://github.com/user-attachments/assets/e7ca1817-dfd2-4447-97dd-1232d30effe2" />


> All search bars that drop down end up behind the hero slider on my home page.

You have changed the header `sticky-top` class to `position-relative`which removes the `z-index` as well. Add `z-1` class to your `header` > `position-relative bg-primary shadow-sm z-1` and it will work.

> Also, instead of updating the page with the filtered products it reloads the page and leaves it blank with no filter or products (though it does show total search results).

The result content is empty. Have you tried this plugin with another theme?


<img width="1357" alt="Bildschirmfoto 2024-12-14 um 13 38 48" src="https://github.com/user-attachments/assets/d1d9212b-f8a1-463b-ba71-cbf2b3032743" />

**Reply by goclimbing** [reply]
Thanks for all of that. I really appreciate you getting back to me quickly.  I logged almost 14 hours yesterday at work migrating our main dev site to this plugin. And remember I only discovered the theme 9 days ago.

Regarding the slider z-indexing.

> You have changed the header sticky-top class to position-relative

Yeah I just went with this example so I could control bg color. 
https://bootscore.me/documentation/filters/#position-and-bg
adding z-1 did indeed fix.  Or I might just want sticky-top - didn't realize I was changing that.

---

Regarding mobile logo. I already had a need to copy header.php over to add the ticker so I'll look at adding code there. Makes sense. 

Mobile menu bg - I feel silly having to ask that.

---
RE - non-functional Product Filter by WBW:

Here's a link to the site before I switched themes. This one is running Understrap.  The filter works fine there as well as with Twenty Twenty-five etc.  It has over 50k installs and a 5 star rating.

My guess that the classes were changed so the filter somehow doesn't target the page correctly.  It reloads the page rather than just updating in page as seen on stg.

https://stg-psbullion-dev.kinsta.cloud/shop/

This does lead me to another question.  I found how to move the right column to the left with the filter.

The product filter has a setting where i can select what pages to show it on. It's set to Shop page only. It's a short code added to that sidebar-1 widget.  The problem is pages like this:
https://psbullion.kinsta.cloud/product/1-fv-40-silver-kennedy-half-dollars-1965-1969-bu/
still display the widget area because it's populated with that short code.  How might I set pages like that to full width?

I thought about coding around this to test is_cart() is_product() and return col sizes based on that.
https://bootscore.me/documentation/filters/#sidebar-to-left

---

RE the Ticker at the top - it seems to be broken everywhere so disregard.

Again, 

Thanks for all the help.

**Reply by crftwrk** [reply]
This is much info packed in one thread, think we should break this down and open a new discussion for each issue in future. 


> ...just want sticky-top - didn't realize I was changing that.

The `sticky-top` class has built-in `z-index` while other positions have not and the search-collapse is inside the header. Usually this is not a problem, but it seems that Smartslider has an higher `z-index` than 0. That's why in your case you have to increase `z-index` for your header if you are removing `sticky-top` class:

| Sticky  | Not sticky |
| ------------- | ------------- |
| <img width="141" alt="sticky-top" src="https://github.com/user-attachments/assets/f3eddbc2-3262-482c-9469-7f250afb711e" />  | <img width="222" alt="relative" src="https://github.com/user-attachments/assets/b132e7a7-9195-4eeb-9f2f-29262033cd51" />  |

> Regarding mobile logo. I already had a need to copy header.php...

Duplicate this snippet:

https://github.com/bootscore/bootscore/blob/50c871b0198b580c3b14fa48e1053db61bca6010/header.php#L47-L51

Then change the lg-logo `navbar-brand` into `navbar-brand d-none d-lg-block` and the mobile one into `navbar-brand d-lg-none` and change paths to logos.

> RE - non-functional Product Filter by WBW:

This is an issue and should be solved. I do not see any error in console, so I think this is an PHP bug. Try following to locate it:

- Disable AJAX cart with this two filters as described here https://bootscore.me/documentation/woocommerce/#restoring-woocommerce-s-default-functionality
- Comment each line in `functions.php` and `wc-functions.php` until you have found it:


https://github.com/bootscore/bootscore/blob/50c871b0198b580c3b14fa48e1053db61bca6010/functions.php#L37-L61

https://github.com/bootscore/bootscore/blob/50c871b0198b580c3b14fa48e1053db61bca6010/woocommerce/wc-functions.php#L18-L29

> because it's populated with that short code. How might I set pages like that to full width?

If it's a none WC page, you can simply select `page-sidebar-none` template. If it's a WC page, then you can't do that. The sidebar is dynamic if there are no widgets active, the content is full width. So, you have set your sidebar widgets conditionally.

For now, there is only one plugin available that is able to show/hide GB widgets depending on conditions https://wordpress.org/plugins/widget-options/. This is an overloaded plugin but works. This is also active on bootscore.me and you can check sidebar on pages with different sidebars:

- https://bootscore.me/archives/category-default/
- https://bootscore.me/shop/
- https://bootscore.me/shop/plugins/bs-dark-mode/ (WC page, no widget active in sidebar)

**Reply by goclimbing** [reply]
@crftwrk Thanks again for all that. Agreed this should get split out. There's a good chance most of this has been addressed and I can take care of it and not need further help.  

>> RE - non-functional Product Filter by WBW:

>This is an issue and should be solved. 

Cool I'll do the work on my end and will create a new thread on that topic alone.  Figuring I can probably locate the culprit but might not determine best way to fix.
